python-package/lightgbm/basic.py

```diff
@@ -727,6 +727,36 @@ def _free_handle(self):
             self.handle = None
         return self
 
+    def _set_init_score_by_predictor(self, predictor, data, used_indices=None):
+        data_has_header = False
+        if isinstance(data, string_type):
+            # check data has header or not
+            if str(params.get("has_header", "")).lower() == "true" \
+                    or str(params.get("header", "")).lower() == "true":
+                data_has_header = True
+        init_score = predictor.predict(data,
+                                       raw_score=True,
+                                       data_has_header=data_has_header,
+                                       is_reshape=False)
+        if used_indices is not None:
+            assert not self.need_slice
+            if isinstance(data, string_type):
+                sub_init_score = [0 for _ in range(self.num_data() * predictor.num_class)]
+                assert self.num_data() == len(used_indices)
+                for i in range_(len(used_indices)):
+                    for j in range_(predictor.num_class):
+                        sub_init_score[i * redictor.num_class + j] = init_score[used_indices[i] * redictor.num_class + j]
+                init_score = sub_init_score
+        if predictor.num_class > 1:
+            # need re group init score
+            new_init_score = np.zeros(init_score.size, dtype=np.float32)
+            num_data = self.num_data()
```

Comment:
I think number of calls `self.num_data()` can be decreased by calculating `num_data = self.num_data()` at the very begging of the method.